scenario-compiler/src/main/scala/pl/touk/nussknacker/engine/extension/ToMapConversionExt.scala

```diff
@@ -94,12 +95,29 @@ object ToMapConversion extends Conversion[JMap[_, _]] {
     case _                 => false
   }
 
-  private def canConvertToMap(c: JCollection[_]): Boolean = c.isEmpty || c
+  private def canConvertToMap(c: JCollection[_]): Boolean = c
     .stream()
     .allMatch {
-      case m: JMap[_, _] => m.keySet().containsAll(keyAndValueNames)
-      case _             => false
+      case m: JMap[_, _]                => m.keySet().containsAll(keyAndValueNames)
+      case _: JMap.Entry[_, _] => true
+      case _                            => false
+    }
+
+  private def convertToMap(c: JCollection[_]): Option[JMap[_, _]] = {
+    val map = new JHashMap[Any, Any]()
+    var foundError = false
+    c.forEach {
+      case e: JMap[_, _]                => map.put(e.get(keyName), e.get(valueName))
+      case e: JMap.Entry[_, _] => map.put(e.getKey, e.getValue)
+      case _ => foundError = true
+    }
+    if (foundError) {
+      // internal error of this class, should not have been called with such value
```

Comment:
I would remove this comment and just simply throw exception (if "it should not happened") because this mutation and handling it only obfuscates this code